omnisafe/algorithms/model_based/base/pets.py

```diff
@@ -238,33 +247,25 @@ def _update_policy(
 
     def _update_dynamics_model(
         self,
-        current_step: int,  # pylint: disable=unused-argument
     ) -> None:
-        """Update dynamics.
-
-        Args:
-            current_step (int): current step.
-        """
+        """Update dynamics."""
         state = self._dynamics_buf.data['obs'][: self._dynamics_buf.size, :]
         action = self._dynamics_buf.data['act'][: self._dynamics_buf.size, :]
         reward = self._dynamics_buf.data['reward'][: self._dynamics_buf.size]
         cost = self._dynamics_buf.data['cost'][: self._dynamics_buf.size]
         next_state = self._dynamics_buf.data['next_obs'][: self._dynamics_buf.size, :]
         delta_state = next_state - state
-        if torch.is_tensor(delta_state):
-            inputs = torch.cat((state, action), -1)
-            inputs = torch.reshape(inputs, (inputs.shape[0], -1))
-
-            labels = torch.reshape(delta_state, (delta_state.shape[0], -1))
-            if self._cfgs.dynamics_cfgs.predict_reward:
-                labels = torch.cat(((torch.reshape(reward, (reward.shape[0], -1))), labels), -1)
-            if self._cfgs.dynamics_cfgs.predict_cost:
-                labels = torch.cat(((torch.reshape(cost, (cost.shape[0], -1))), labels), -1)
-            inputs = inputs.cpu().detach().numpy()
-            labels = labels.cpu().detach().numpy()
-        assert not torch.is_tensor(inputs) and not torch.is_tensor(
-            labels,
-        ), 'inputs and labels should be numpy array'
+        assert isinstance(delta_state, torch.Tensor), 'delta_state should be torch.Tensor'
+        inputs = torch.cat((state, action), -1)
+        inputs = torch.reshape(inputs, (inputs.shape[0], -1))
+
+        labels = torch.reshape(delta_state, (delta_state.shape[0], -1))
+        if self._cfgs.dynamics_cfgs.predict_reward:
+            labels = torch.cat(((torch.reshape(reward, (reward.shape[0], -1))), labels), -1)
+        if self._cfgs.dynamics_cfgs.predict_cost:
+            labels = torch.cat(((torch.reshape(cost, (cost.shape[0], -1))), labels), -1)
+        inputs = inputs.cpu().detach().numpy()
+        labels = labels.cpu().detach().numpy()
```

Comment:
If the `delta_state` is not a tensor, something went wrong. So I think `assert` is better than `if` here.